File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        HashMap<Character, Character> maps = new HashMap<>();
        maps.put(')','(');
        maps.put(']','[');
        maps.put('}','{');
        Stack<Character> stack = new Stack<>();
        for (char ch: s.toCharArray()){
            if (!maps.containsKey(ch)){
                stack.push(ch);
            }else{
                 if(stack.empty() || stack.pop() != maps.get(ch) )
                    return false;   
            }
        }
        return stack.empty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        valid_pairs = { '(' : ')', '{' : '}', '[' : ']' }
        open_paren = set(['(', '{', '['])
        my_stack = []

        for ch in s:
            if ch in open_paren:
                my_stack.append(ch)
            elif my_stack and ch == valid_pairs[my_stack[-1]]:
                my_stack.pop()
            else:
                return False
        return my_stack == []